### PR TITLE
{devel}[foss/2016b] TensorFlow v1.0.1, SWIG v3.0.10, wheel v0.30.0a0, ...

### PR DESCRIPTION
**Pull request by gppezzi**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/s/SWIG/SWIG-3.0.10-foss-2016b-Python-3.5.2.eb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+
+dependencies = [
+    ('Python', '3.5.2'),
+    ('PCRE', '8.38'),
+]
+
+# fix for TensorFlow
```
Comment (gppezzi) [inline easybuild/easyconfigs/s/SWIG/SWIG-3.0.10-foss-2016b-Python-3.5.2.eb]: This was an issue on Cray  and using older version of Bazel, I've just removed the fix and it seems to work now.

File: easybuild/easyconfigs/s/SWIG/SWIG-3.0.10-foss-2016b-Python-3.5.2.eb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# @author: gppezzi, marcelsc
+
+name = 'SWIG'
+version = '3.0.10'
+pyver = '3.5.2'
```
Comment (gppezzi) [inline easybuild/easyconfigs/s/SWIG/SWIG-3.0.10-foss-2016b-Python-3.5.2.eb]: done

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.0.0-foss-2016b-Python-3.5.2.eb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+files_to_copy = [
+	      (['%(builddir)s/tensorflow-%(version)s-cp35-cp35m-linux_x86_64.whl'], '%(installdir)s'),
+	      (['%(builddir)s/bin'], '%(installdir)s'),
+	      (['%(builddir)s/lib'], '%(installdir)s'),
+]
+
+sanity_check_paths={
+    'files': ['%(builddir)s/tensorflow-%(version)s-cp35-cp35m-linux_x86_64.whl'],
+    'dirs': ['%(installdir)s/lib/python%(pyshortver)s/site-packages'],
```
Comment (gppezzi) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.0.0-foss-2016b-Python-3.5.2.eb]: done

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.0.0-foss-2016b-Python-3.5.2.eb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+]
+
+files_to_copy = [
+	      (['%(builddir)s/tensorflow-%(version)s-cp35-cp35m-linux_x86_64.whl'], '%(installdir)s'),
+	      (['%(builddir)s/bin'], '%(installdir)s'),
+	      (['%(builddir)s/lib'], '%(installdir)s'),
+]
+
+sanity_check_paths={
+    'files': ['%(builddir)s/tensorflow-%(version)s-cp35-cp35m-linux_x86_64.whl'],
```
Comment (gppezzi) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.0.0-foss-2016b-Python-3.5.2.eb]: No idea, I'll blame one of the co-authors 😄 
Fixed.

File: easybuild/easyconfigs/t/TensorFlow/tensorflow-v1.0.0-Python-3.5.2-foss.patch [hunk 5]
```diff
@@ -0,0 +1,103 @@
+diff --git a/configure-cscs.sh b/configure-cscs.sh
```
Comment (gppezzi) [inline easybuild/easyconfigs/t/TensorFlow/tensorflow-v1.0.0-Python-3.5.2-foss.patch]: done

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.0.0-foss-2016b-Python-3.5.2.eb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+]
+
+sources = ['v%(version)s.tar.gz']
+source_urls = ['https://github.com/tensorflow/tensorflow/archive/']
+
+
+# Patch to remove hardcoded paths for GCC and Binutils
+patches = [('tensorflow-v%(version)s-Python-%(pyver)s-foss.patch')]
+
+with_configure = False
```
Comment (gppezzi) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.0.0-foss-2016b-Python-3.5.2.eb]: does this have any effect on a CmdCp easyblock?

**Comment by gppezzi** [comment]
Test report by @gppezzi
**SUCCESS**
Build succeeded for 5 out of 5 (4 easyconfigs in this PR)
leone2 - Linux RHEL 6.7, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/1eefb0944a837f2672ec3fa5be9a4b10 for a full test report.

File: easybuild/easyconfigs/t/TensorFlow/tensorflow-v1.0.0-Python-3.5.2-foss.patch [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+new file mode 100644
+index 0000000..799fdbf
+--- /dev/null
++++ b/configure-cscs.sh
+@@ -0,0 +1,23 @@
++#!/bin/bash
++
++sed -i s\@'$EBROOTGCCCORE'@"$EBROOTGCCCORE"@g third_party/gpus/crosstool/CROSSTOOL.tpl
++sed -i s\@'$EBROOTBINUTILS'@"$EBROOTBINUTILS"@g third_party/gpus/crosstool/CROSSTOOL.tpl
++sed -i s\@'$EBROOTGCCCORE'@"$EBROOTGCCCORE"@g third_party/gpus/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc.tpl
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorFlow/tensorflow-v1.0.0-Python-3.5.2-foss.patch]: +1, but thanks for sharing @gppezzi, this is going to be a major help!

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.0.0-foss-2016b-Python-3.5.2.eb (hunk 6 above)
Comment (boegel) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.0.0-foss-2016b-Python-3.5.2.eb]: well, `CmdCp` derives from `MakeCp`, so it's taken into account, but the default value is already `False`:

```
$ eb -a -e CmdCp | grep with_configure
with_configure*         Run configure script before building [default: False]
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 33 out of 33 (4 easyconfigs in this PR)
node2520.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/75dbaa98a94f78994c167df3574033fe for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2156.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/3e20876b4d8d14d680e5821b37ab3d2d for a full test report.

File: easybuild/easyconfigs/t/TensorFlow/tensorflow-v1.0.0-Python-3.5.2-foss.patch (hunk 7 above)
Comment (gppezzi) [inline easybuild/easyconfigs/t/TensorFlow/tensorflow-v1.0.0-Python-3.5.2-foss.patch]: I'm not against, but I don't have time to work on this :)

**Comment by RvDijk** [comment]
What is the status on this PR?
Also, is it possible that the protobuf-python module also needs to be loaded in order for this to work?

**Comment by gppezzi** [comment]
Status on this PR: the reviewers suggested to write a custom easyblock for fixing the hardcoded absolute paths (instead of [this hack](https://github.com/hpcugent/easybuild-easyconfigs/pull/4412/files#diff-52c9bdec2e4eb51af97d98cd4cb7bc7bR14)). 

I'm not planning to work on this anytime soon, so PRs against my branch are welcome.

@RvDijk did you have problems because of missing protobuf? 

I've removed it from the dependency list because Bazel was ignoring it and building a new one inside its own workspace (see fix for protobuf [here](https://github.com/hpcugent/easybuild-easyconfigs/pull/4412/files#diff-b8f1de7cb3c7e9e3e761ce3057ee37f0R43)).

**Comment by RvDijk** [comment]
@gppezzi leaving out the protobuf-python module gives me the following when importing the tensorflow module in Python (after having succesfully installed this tensorflow eb):
```
>>> import tensorflow as tf
Traceback (most recent call last):
File "<stdin>", line 1, in <module>
File "/software/software/tensorflow/1.0.1-foss-2016a-Python-3.5.2/lib/python3.5/site-packages/tensorflow/__init__.py", line 24, in <module>
from tensorflow.python import *
File "/software/software/tensorflow/1.0.1-foss-2016a-Python-3.5.2/lib/python3.5/site-packages/tensorflow/python/__init__.py", line 75, in <module>
from tensorflow.core.framework.graph_pb2 import *
File "/software/software/tensorflow/1.0.1-foss-2016a-Python-3.5.2/lib/python3.5/site-packages/tensorflow/core/framework/graph_pb2.py", line 6, in <module>
from google.protobuf import descriptor as _descriptor
ImportError: No module named 'google'
```
And in this case including protobuf-python (3.2.0) fixed it. Could be a local issue if this does not occur anywhere else. Does it work for you without it?

**Comment by gppezzi** [comment]
@RvDijk thanks for reporting this, I'm including protobuf-python as dependency. 

Maybe we should write a comment mentioning that this protobuf-python version is not the one used for building TF but it is needed at runtime?  

Unless someone wants to find a way to re-use the contents of the bazel workspace 😄

**Comment by RvDijk** [comment]
@gppezzi it would be nice to mention it, otherwise it might confuse others.
I wonder if figuring out the bazel workspace for re-use is worth the time... 😃

**Comment by boegel** [comment]
This may also be useful: https://github.com/truatpasteurdotfr/docker-centos6-tensorflow/blob/master/runme.sh

**Comment by boegel** [comment]
More useful information for building Tensorflow from source with Intel compilers: https://software.intel.com/en-us/articles/tensorflow-optimizations-on-modern-intel-architecture

**Comment by boegel** [comment]
More: https://gist.github.com/taylorpaul/3e4e405ffad1be8355bf742fa24b41f0

**Comment by boegel** [comment]
yet another one: https://github.com/mkandes/naked-singularity/tree/master/definition-files/us/ucsd/sdsc/comet/tensorflow

**Comment by boegel** [comment]
Finally made some progress here... First step: an easyblock for `Bazel`, see https://github.com/easybuilders/easybuild-easyblocks/pull/1286, plus accompanying easyconfig file for latest Bazel version, see https://github.com/easybuilders/easybuild-easyconfigs/pull/5311.

**Comment by bartoldeman** [comment]
notes:
https://wiki.calculquebec.ca/w/Tensorflow/en
see
https://wiki.calculquebec.ca/w/Tensorflow_v1.2.x

**Comment by boegel** [comment]
@gppezzi We now have a custom easyblock for TensorFlow, see https://github.com/easybuilders/easybuild-easyblocks/pull/1287, and easyconfigs for building TensorFlow from source, both CPU-only (see #5819) and with GPU support (see #5769).

So this can be closed?

**Comment by gppezzi** [comment]
Sure.

**Pull request closed without merging**